bip-0066.mediawiki

```diff
@@ -55,24 +55,30 @@ bool static IsValidSignatureEncoding(const std::vector<unsigned char> &sig) {
     // Make sure the length covers the entire signature.
     if (sig[1] != sig.size() - 3) return false;
 
+    // Check whether the R element is an integer.
+    if (sig[2] != 0x02) return false;
+
     // Extract the length of the R element.
     unsigned int lenR = sig[3];
 
-    // Make sure the length of the S element is still inside the signature.
+    // Zero-length integers are not allowed for R.
+    if (lenR == 0) return false;
+
+    // Make sure the length of the R element is still inside the signature.
```

Comment:
This is true,  though I saw the `Status: draft` and thought it would be ok at least bring it up.

As you said, it is only slightly,  however it changes the order from:
```
  # STRICT BITCOIN FORMAT
// Minimum and maximum size constraints.

  # DER FORMAT/LENGTH
  # 0x02 INTEGERS ASSUMED
// A signature is of type 0x30 (compound).
// Make sure the length covers the entire signature.
// Extract the length of the R element.
// Make sure the length of the R element is still inside the signature.
// Extract the length of the S element.
// Verify that the length of the signature matches the sum of the length of the elements.

   # R DER FORMAT
// Check whether the R element is an integer.
// Zero-length integers are not allowed for R.

   # R STRICT BITCOIN FORMAT
// Negative numbers are not allowed for R.
// Null bytes at the start of R are not allowed, unless R would otherwise be interpreted as a negative number.

   # S DER FORMAT
// Check whether the S element is an integer.
// Zero-length integers are not allowed for S.

   # S STRICT BITCOIN FORMAT
// Negative numbers are not allowed for S.
// Null bytes at the start of S are not allowed, unless S would otherwise be interpreted as a negative number.
```

To

```
        # STRICT BITCOIN FORMAT
    // Minimum and maximum size constraints.

        # DER FORMAT
    // A signature is of type 0x30 (compound).
    // Make sure the length covers the entire signature.

            # R DER FORMAT
        // Check whether the R element is an integer.
        // Extract the length of the R element.
        // Zero-length integers are not allowed for R.
        // Make sure the length of the R element is still inside the signature.

            # S DER FORMAT
        // Check whether the S element is an integer.
        // Extract the length of the S element.
        // Zero-length integers are not allowed for S.
        // Verify that the length of the signature matches the sum of the length of the elements.

         # STRICT BITCOIN FORMAT
    // Negative numbers are not allowed for R.
    // Null bytes at the start of R are not allowed, unless R would otherwise be interpreted as a negative number.
    // Negative numbers are not allowed for S.
    // Null bytes at the start of S are not allowed, unless S would otherwise be interpreted as a negative number.
```